packages/jest-haste-map/src/worker.js

```diff
@@ -49,39 +30,34 @@ module.exports = (data: WorkerMessage, callback: WorkerCallback): void => {
       throw new Error('jest-haste-map: hasteImplModulePath changed');
     }
     hasteImplModulePath = data.hasteImplModulePath;
-    hasteImpl =
-      // $FlowFixMe: dynamic require
-      (require(hasteImplModulePath): HasteImpl);
+    // $FlowFixMe: dynamic require
+    hasteImpl = (require(hasteImplModulePath): HasteImpl);
   }
 
-  try {
```

Comment:
Just to double check, when a worker call throws, `jest-worker` handles it differently than the process crashing, right? Throwing will *not* result in a retry, just in a rejected Promise in the parent.

Reply:
Exactly. A `throw` on your worker module will cause a rejected promise on the parent process, but it will not retry anything; since the retry mechanism is tied to `childProcess.on('exit')`.